config/flat.yaml

```diff
@@ -0,0 +1,80 @@
+# This is an example of how to make a flat-field image with imsim.
+
+
+# Load the imsim modules.
+modules:
+    - imsim
+
+input:
+
+    tree_rings:
+        # This enables TreeRingCenter and TreeRungFunc, which are stored as a dict here based
+        # on the detector name, so the right value can be accessed for each object.
+        # This file lives in the imsim repo.
+        file_name: "tree_ring_data/tree_ring_parameters_2018-04-26.txt"
+        # Can limit the detectors to read in.  This is handy while debugging, since it takes
+        # a minute or so to read all 189 detectors (the default behavior).
+        only_dets: [R22_S11]
+
+
+image:
+    type: LSST_Flat
+
+    random_seed: 1234
+
+    xsize: 4096
+    ysize: 4096
+
+    counts_per_pixel: 5000
+
+    wcs:
+        type: Batoid
+
+        # These are required, but the actual values don't matter for a flat field.
+        boresight:
+            type: RADec
+            ra: "0 degrees"
+            dec: "-30 degrees"
+        rotTelPos: "0 degrees"
+        obstime: "2024-01-01"
+
+        # These are relevant for a flat, so should make sure to get these right.
+        det_name: $det_name
+        band: r
+
+    noise:
+        type: Poisson
+
+    sensor:
+        type: Silicon
+        strength: 1.0
+        treering_center: { type: TreeRingCenter, det_name: $det_name }
+        treering_func: { type: TreeRingFunc, det_name: $det_name }
+
+
+# This defines both the output files and some basic things about the overall exposure/fov.
+output:
+    type: LSST_CCD
+    nproc: 1    # Change this to work on multiple CCDs at once.
+    nfiles: 1   # Default is all 189 CCDs.  Set to 1 while testing.
+
+    camera: LsstCam
+
+    exp_time: 30
+
+    cosmic_ray_rate: 0.2
+
+    det_num:
+        type: Sequence
+        nitems: 189
+        first: 94  # Can set first to something if you want to do a specific sensor.
+
+    dir: flats
+    file_name:
+        type: FormattedStr
+        format : flat_%05d-%s-%s-det%03d.fits
+        items:
+            - 1  # When making multiple flats, probably should increment this for each file.
+            - "@image.wcs.band"
+            - "$det_name"   # A value stored in the dict by LSST_CCD
+            - "@output.det_num"
```

Comment:
To make use of the flats by the LSST code, they'll need to be formatted into FITS files with per-amp HDUs, just like the on-sky images.   So we should include a `readout` section in this example.  However, that requires information from the `OpsimMetaDict` class.  This aligns with the needs of DM code to be able to ingest calibration data.  One way to handle this would be to have an opsim db specifically for making calibration frames.

Reply:
I just tried to ingest the per-amp file into a test repo using the LSST code.  Unfortunately, it complains that it can't process the file, but it doesn't say what's missing that it expects.   The on-sky per-amp images can be ingested, so there's some subset of header keywords in those files, e.g., `RUNNUM`, that are needed for ingesting the flats.  I think we should go ahead and merge this branch and sort out with the DM folks later what keywords are needed for these files.